### PR TITLE
Add Library Content Provider

### LINKED ISSUE #7: Add library content provider
Explore if library content provider could be useful for this package.

### PR DESCRIPTION
**Pull request by swif2**
This closes https://github.com/mkj-is/PathBuilder/issues/7.

### CODE REVIEW

File: Sources/PathBuilder/LibraryContent.swift [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import SwiftUI
+import DeveloperToolsSupport
+
+@available(macOS 11.0, *)
```
Comment (mkj-is) [inline Sources/PathBuilder/LibraryContent.swift]: This library is available an all the platform. To be able to compile it we need to add checks on all of them.

```suggestion
@available(macOS 11.0, iOS 14.0, tvOS 14.0, watchOS 7.0, *)
```

File: Sources/PathBuilder/LibraryContent.swift [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+struct LibraryContent: LibraryContentProvider {
+    var views: [LibraryItem] {
+        LibraryItem(
+            Path {
+                Move(to: CGPoint(x: 0, y: 0))
+                // ...
+                Close()
+            }
```
Comment (mkj-is) [inline Sources/PathBuilder/LibraryContent.swift:12]: Compilation is failing in current production version of Xcode (14.3.1). In Xcode 15 beta 4 this compiles.

**Review by mkj-is — COMMENTED** [review]
Thanks for this contribution, can you please check those comments I added?

File: Sources/PathBuilder/LibraryContent.swift (hunk 2 above)
Comment (mkj-is) [reply Sources/PathBuilder/LibraryContent.swift:12]: Screenshot 2023-07-21 at 17 18 02

File: Sources/PathBuilder/LibraryContent.swift [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        )
+        LibraryItem(Arc(center: .zero, radius: 20, startAngle: .zero, endAngle: .degrees(180)))
+        LibraryItem(Close())
+        LibraryItem(Curve(to: CGPoint(x: 10, y: 10), control1: CGPoint(x: 5, y: 5), control2: CGPoint(x: 10, y: 10)))
+        LibraryItem(EmptySubpath())
+        LibraryItem(Line(to: CGPoint(x: 50, y: 50)))
+        LibraryItem(Lines(CGPoint(x: 60, y: 60), CGPoint(x: 70, y: 70)))
+        LibraryItem(Loop(sequence: 0..<10) { i in
+            Line(to: CGPoint(x: i * 5, y: i * 5))
+        })
```
Comment (mkj-is) [inline Sources/PathBuilder/LibraryContent.swift]: Compiler failure in Xcode 14.3.1 is caused by this loop, we can remove it for now.

```suggestion
```

File: Sources/PathBuilder/LibraryContent.swift (hunk 2 above)
Comment (swif2) [reply Sources/PathBuilder/LibraryContent.swift:12]: This weird. I compiled this with no issue on Xcode 14.2. My mac is too old too upgrade to the latest version.

Comment (swif2) [reply Sources/PathBuilder/LibraryContent.swift:12]: No sure if removing the embedded code in the trailing closure will work.

File: Sources/PathBuilder/LibraryContent.swift (hunk 3 above)
Comment (swif2) [reply Sources/PathBuilder/LibraryContent.swift]: Does removing this also fix the issue shown in the snapshot above? Sorry, I never expected it wouldn't work with latest Xcode.

Comment (mkj-is) [reply Sources/PathBuilder/LibraryContent.swift]: Yes, this fixes all the compiler issues.

File: Sources/PathBuilder/LibraryContent.swift (hunk 2 above)
Comment (mkj-is) [reply Sources/PathBuilder/LibraryContent.swift:12]: This fixes the warning: https://github.com/mkj-is/PathBuilder/pull/8/files#r1270831463

**Comment by mkj-is** [comment]
Thank you, I will release a new version sometime next week.

**Comment by swif2** [comment]
Thank you😉

**Pull request merged**